backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java

```diff
@@ -340,4 +344,46 @@ private static ResourceVisibility getResourceVisibility(
     }
     return ResourceVisibility.PRIVATE;
   }
+
+  private static void deleteRoleBindingsForProjects(List<ProjectEntity> projectEntities) {
+    // set roleBindings name by accessible projects
+    List<String> roleBindingNames = new LinkedList<>();
+    setRoleBindingsNameOfAccessibleProjectsInRoleBindingNamesList(
+        projectEntities, roleBindingNames);
+    LOGGER.debug("num bindings after Projects {}", roleBindingNames.size());
+
+    // Remove all role bindings
+    if (!roleBindingNames.isEmpty()) {
+      roleService.deleteRoleBindings(roleBindingNames);
+    }
+  }
+
+  private static void setRoleBindingsNameOfAccessibleProjectsInRoleBindingNamesList(
+      List<ProjectEntity> allowedProjects, List<String> roleBindingNames) {
+    for (ProjectEntity project : allowedProjects) {
+      String projectId = project.getId();
+
+      String ownerRoleBindingName =
+          roleService.buildRoleBindingName(
+              ModelDBConstants.ROLE_PROJECT_OWNER,
+              project.getId(),
+              project.getOwner(),
+              ModelDBServiceResourceTypes.PROJECT.name());
+      if (ownerRoleBindingName != null) {
+        roleBindingNames.add(ownerRoleBindingName);
+      }
+
+      // Delete workspace based roleBindings
+      List<String> workspaceRoleBindingNames =
+          roleService.getWorkspaceRoleBindings(
+              project.getWorkspace(),
+              WorkspaceTypeEnum.WorkspaceType.forNumber(project.getWorkspace_type()),
+              projectId,
+              ModelDBConstants.ROLE_PROJECT_ADMIN,
+              ModelDBServiceResourceTypes.PROJECT,
+              project.getProjectVisibility().equals(VisibilityEnum.Visibility.ORG_SCOPED_PUBLIC),
+              "_GLOBAL_SHARING");
+      roleBindingNames.addAll(workspaceRoleBindingNames);
+    }
+  }
```

Comment:
it is code restored from DeleteEntitiesCron, but without collaborator bindings deletion